frontend/src/static/js/v_summary.js

```diff
@@ -792,12 +808,51 @@ window.vSummary = {
     getGroupTotalContribution(group) {
       return group.reduce((accContribution, user) => accContribution + user.totalCommits, 0);
     },
+
+    restoreZoomFiltered(info) {
+      const {
+        repoIndex, userIndex, filterGroupSelection, isMergeGroup, filterTimeFrame, zoomSince,
+        zoomUntil, sortingOption, sortingWithinOption, isSortingDsc, isSortingWithinDsc,
+      } = info;
+      let filtered = [];
+
+      const groups = JSON.parse(JSON.stringify(this.repos));
```

Comment:
>The overhead I mean is that, our repo needs to be filtered twice to produce the extra setting for v_zoom, which will be a lot of time when there are many repos.

I see, do you mean this part? Then yes, its not quite okay because of the additional parsing.
Also, the merge group should be more targeted and not perform on all the repos.

In short, would prefer as little processing as possible to be performed. In fact, can even consider storing indexes of the repos in merged group instead so that no filtering is needed.

Reply:
@fzdy1914 @eugenepeh 

I have modified such that only the repos that are involved in the zoom tab will be processed. 

The zoom tab hashes and params are also shortened.